script-runner/AutomatedTestRunner.hs

```diff
@@ -0,0 +1,400 @@
+{-# LANGUAGE ApplicativeDo   #-}
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DeriveGeneric   #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving   #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
```

Comment:
Probably want to drop that extension.

In some situations its fine, but in others its a real hinderance to readability and understandability and there is no way to tell the compiler to allow the relatively benign uses and disallow the bad uses.